### PR TITLE
PayPal - remove `offerCredit` and move `userAction` to checkout request

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

- Remove `offerCredit` from BTPayPalRequest. @demerino confirmed that this is OK to remove in favor of `offerPayLater`.
- Move `userAction` from BTPayPalRequest to BTPayPalCheckoutRequest. @demerino confirmed that it only applies to checkout flows.
- Update the "PayPal Credit" demo to be a "PayPal Pay Later" demo

### Checklist

- [x] Added a changelog entry

- @sestevens

### CODE REVIEW

File: Demo/Application/Base/Settings/Settings.bundle/Root.plist [hunk 1]
```diff
@@ -19,13 +19,13 @@
 			<array>
 				<string>PayPal - Billing Agreement</string>
 				<string>PayPal - One-Time Payment</string>
```
Comment (sestevens) [inline Demo/Application/Base/Settings/Settings.bundle/Root.plist:21]: At some point we should also update the use of "Billing Agreement" and "One-Time Payment" in the demo app to be "Vault" and "Checkout". But I'm going to leave that for a separate PR, especially since it isn't necessary for the v5 release.

**Review by sarahkoop — APPROVED** [review]
Looks good to me 🌟

File: Demo/Application/Features/PayPal - Pay Later/BraintreeDemoPayPalPayLaterViewController.m [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    [driver tokenizePayPalAccountWithPayPalRequest:request completion:^(BTPayPalAccountNonce * _Nullable payPalAccount, NSError * _Nullable error) {
+        [sender setEnabled:YES];
+
+        if (error) {
+            self.progressBlock(error.localizedDescription);
+        } else if (payPalAccount) {
+            self.completionBlock(payPalAccount);
+        } else {
+            self.progressBlock(@"Cancelled");
```
Comment (hollabaq86) [inline Demo/Application/Features/PayPal - Pay Later/BraintreeDemoPayPalPayLaterViewController.m:37]:
```suggestion
            self.progressBlock(@"Canceled");
```

Comment (hollabaq86) [reply Demo/Application/Features/PayPal - Pay Later/BraintreeDemoPayPalPayLaterViewController.m:37]: Absolute nit of a comment, here

**Review by hollabaq86 — APPROVED** [review]
:shipit:

**Comment by sestevens** [comment]
The one failing UI test (`testThreeDSecurePaymentFlowV1_unavailable`) is passing locally, so we're going to go ahead and merge this.
